graphql_compiler/tests/integration_tests/test_backends_integration.py

```diff
@@ -555,6 +555,112 @@ def test_recurse(self) -> None:
+                    {"animal": "Animal 1", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 1", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 2", "ancestor": "Animal 2"},  # self
+                    {"animal": "Animal 2", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 2", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 3", "ancestor": "Animal 3"},  # self
+                    {"animal": "Animal 3", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 3", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 4", "ancestor": "Animal 4"},  # self
+                    {"animal": "Animal 4", "ancestor": "Animal 3"},  # parent
+                    {"animal": "Animal 4", "ancestor": "Animal 1"},  # grandparent
+                ],
+            ),
+            # Query 4: Unfold recursion to omit self
+            (
+                """
+                {
+                    Animal {
+                        name @output(out_name: "animal")
+                        in_Animal_ParentOf {
+                            in_Animal_ParentOf @recurse(depth: 1) {
+                                name @output(out_name: "ancestor")
+                            }
+                        }
+                    }
+                }""",
+                [
+                    {"animal": "Animal 1", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 1", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 2", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 2", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 3", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 3", "ancestor": "Animal 1"},  # grandparent
+                    {"animal": "Animal 4", "ancestor": "Animal 1"},  # parent
+                    {"animal": "Animal 4", "ancestor": "Animal 3"},  # grandparent
+                ],
+            ),
+        ]
+
+        # TODO(bojanserafimov): Only testing in MSSQL because none of our backends agree on recurse
+        #                       semantics when multiple paths to the same output are inolved:
+        #                       - Our Match backend would represent each result once, even though it
+        #                         was reached multiple times by different paths.
+        #                       - Our SQL backend would duplicate the output row once for each path
+        #                       - Our Neo4j backend would find all different paths that use each
+        #                         edge at most once, and duplicate the result for each one.
+        for graphql_query, expected_results in queries:
+            self.assertResultsEqual(graphql_query, parameters, test_backend.MSSQL, expected_results)
```

Comment:
Can we check this in Postgres as well? I expect our CTE setup should behave equivalently across SQL implementations, but N=2 will make me a lot more confident in that statement than N=1 😄

Reply:
Postgres is not configured for integration testing, and it's not trivial to do so (different db + schema hierarchy)